File: src/praktikum3/kegiatan2Integer/CoffeeShop.java
/**
 @author: rizkyhaksono
 */

package praktikum3.kegiatan2Integer;

public class CoffeeShop {

    public FirstLastLink theList;

    public CoffeeShop() {
        theList = new FirstLastLink();
    }

    public void enqueue(int j) {
        theList.insertLast(j);
    }

    public long dequeue() {
        return theList.deleteFirst();
    }

    public void peek() {
        System.out.println("Top stack : " + theList.first.dData);
    }

    public boolean isEmpty() {
        return theList.isEmpty();
    }

    public void size() {
        int temp = theList.first.dData + theList.last.dData;
        System.out.println("Size : " + temp + "\n");
    }

    public void displayQueue() {
        System.out.println("Queue [Head -> Tail] : ");
        theList.displayLink();
    }

    public static void main(String[] args) {
        CoffeeShop queue = new CoffeeShop();

        queue.enqueue(10);
        queue.enqueue(40);
        queue.displayQueue();
        System.out.println("Empty : " + queue.isEmpty());
        queue.peek();
        queue.size();

        queue.enqueue(50);
        queue.enqueue(5);
        queue.displayQueue();
        System.out.println("Empty : " + queue.isEmpty());
        queue.peek();
        queue.size();

        queue.dequeue();
        queue.displayQueue();
        System.out.println("Empty : " + queue.isEmpty());
        queue.peek();
        queue.size();
    }
}

File: src/praktikum3/kegiatan2Integer/CoffeeShop.py
__author__      = "rizkyhaksono"
__copyright__   = "Copyright 2023, Malang"

class CoffeeShop:
    def __init__(self):
        self.items = []
    def isEmpty(self):
        return self.items == []
    def enqueue(self, item):
        self.items.insert(0,item)
    def dequeue(self):
        return self.items.pop()
    def size(self):
        return len(self.items)
    def show(self):
        return self.items
    def peek(self):
        return self.items[len(self.items)-1]

if __name__ == "__main__":
    q = CoffeeShop()
    q.enqueue(10)
    q.enqueue(20)
    print(q.show())

    q.enqueue(30)
    q.enqueue(50)
    print(q.show())

    print(q.dequeue())
    print(q.show())

    print(q.peek())
    print(q.show())

    print(q.size())
    print(q.show())

    print(q.isEmpty())